### PR TITLE
Changed logic + output + fixed issues

### PR DESCRIPTION
**Pull request by coffeecrank**
This is an update to the existing test. 
1) I created a better distinction between links and also prioritized groups (fail, then warn, then pass).
2) Output looks more clear and has a nice overall statistics printout.
3) Added wget to parse the title easily, whereas curl will be parsing status code and effective url.
4) Fixed the issues where links to the Product Documentation page were considered redirects. I think this came form the fact that anchors were not parsed correctly (e.g. when curling an anchor link, effective url will always be different from original url).

### CODE REVIEW

**Comment by tisnik** [comment]
LGTM, but why the support for forbidden links was removed? IIRC it was requested by the CCS team :-)

**Comment by coffeecrank** [comment]
@tisnik It wasn't. It's right there in the main link check. But since it's a warning, I put it below all fails. :) I'm still not sure about QA links though... and whether failing them is the right thing. :)

**Comment by tisnik** [comment]
@coffeecrank oh that's why I missed it :) TYVM.

**Comment by tisnik** [comment]
@coffeecrank I'd suggest to merge this pull requests and prepare the second one with changes in error reporting for QA links. We can mark this new PR with 'WIP' and merge it only when our users decide they wanted to process QA links differently.

**Comment by coffeecrank** [comment]
@tisnik Sounds like a plan!

File: TechnicalAccuracy.lua [hunk 1]
```diff
@@ -270,6 +250,7 @@ function TechnicalAccuracy:checkLinks(links, message)
     for _, link in ipairs(links) do
         local pageTitle = execCaptureOutputAsString(wgetCommand(link))
         local redirectAndStatusCode = execCaptureOutputAsTable(curlCommand(link))
+        print(redirectAndStatusCode[2])
```
Comment (tisnik) [inline TechnicalAccuracy.lua]: leftover?

**Comment by tisnik** [comment]
(are you going to squash the last 7 commits? :)

**Comment by coffeecrank** [comment]
@tisnik Yeah, apologies, should be working on another branch probably. :) Was testing QA links, and it seems that I get a 200 response from them on Jenkins, but the body of the response (page itself) is empty. I even tried different RegEx (Lua native), but still no luck. The last commit is final though, we just gotta figure how to set correct permission on Jenkins job.

**Comment by tisnik** [comment]
It looks good. Thank you and let's merge those changes :)

**Pull request merged**